### PR TITLE
Feature/bookmark

### PR DESCRIPTION
**Pull request by thunyathep**
@Balloii @aofattaporn @teamchz @sura1542

### CODE REVIEW

**Review by sura1542 — APPROVED** [review]

File: lib/src/blocs/post/post_bloc.dart [hunk 1]
```diff
@@ -189,6 +189,65 @@ class PostBloc extends Bloc<PostEvent, PostState> {
...
+          if(listPostModel.posts[i].postId == event.postId){
+            index = i;
+            break;
+          }
+        }
+        listPostModel.posts[index].isBookmark = true;
+        await postRepository.AddBookMark(event.postId);
+        print("Added leaw");
+      } catch (e, stacktrace) {
+        print("Exxception occured: $e stackTrace: $stacktrace");
```
Comment (aofattaporn) [inline lib/src/blocs/post/post_bloc.dart]: print remove this !

File: lib/src/blocs/post/post_bloc.dart [hunk 2]
```diff
@@ -189,6 +189,65 @@ class PostBloc extends Bloc<PostEvent, PostState> {
...
+      emit(BookMarkLoadingState());
+      try {
+        final listPostJSON = await postRepository.getMyBookMark();
+        final listPostData = listPostModelFromData(listPostJSON);
+        print(listPostJSON);
+
+
+        // print for check each post
+        print("------ in Bloc-------------");
+        listPostData.posts.forEach((element) {
```
Comment (aofattaporn) [inline lib/src/blocs/post/post_bloc.dart]: and this lol

File: lib/src/screens/BookMark.dart [hunk 3]
```diff
@@ -1,14 +1,43 @@
...
+      // required this.postId,
+      // required this.content,
+      // required this.date,
+      // required this.countComment,
+      // required this.countLike,
+      // required this.isLike,
+      // this.tempIsLike,
+      // required this.category,
+      // required this.postBloc
+      })
```
Comment (aofattaporn) [inline lib/src/screens/BookMark.dart]: if you not use, pls to remove

**Review by aofattaporn — COMMENTED** [review]
please to remove something before push again

**Review by teamchz — APPROVED** [review]
LGTM

**Pull request merged**